core/src/test/java/io/kestra/core/services/PluginDefaultServiceTest.java

```diff
@@ -214,34 +223,64 @@ public void prefix() {
             ))
             .build();
 
-        Flow injected = pluginDefaultService.injectDefaults(flow);
+        FlowWithSource injected = pluginDefaultService.injectDefaults(flow);
 
         assertThat(((DefaultTester) injected.getTasks().getFirst()).getSet(), is(666));
         assertThat(((DefaultTester) injected.getTasks().getFirst()).getValue(), is(2));
     }
 
     @Test
     void alias() {
-        DefaultTester task = DefaultTester.builder()
-            .id("test")
-            .type(DefaultTester.class.getName())
-            .set(666)
-            .build();
-
-        Flow flow = Flow.builder()
-            .tasks(Collections.singletonList(task))
+        String source = """
+            id: default-test
+            namespace: io.kestra.tests
+
+            tasks:
+            - id: test
+              type: io.kestra.core.services.PluginDefaultServiceTest$DefaultTester
+              set: 666""";
+
+        FlowWithSource flow = yamlFlowParser.parse(source, Flow.class)
+            .withSource(source)
+            .toBuilder()
             .pluginDefaults(List.of(
                 new PluginDefault("io.kestra.core.services.DefaultTesterAlias", false, ImmutableMap.of(
                     "value", 1
                 ))
             ))
             .build();
 
-        Flow injected = pluginDefaultService.injectDefaults(flow);
+        FlowWithSource injected = pluginDefaultService.injectDefaults(flow);
 
         assertThat(((DefaultTester) injected.getTasks().getFirst()).getValue(), is(1));
     }
 
+    @Test
+    void defaultOverride() {
+        String source = """
+            id: default-test
+            namespace: io.kestra.tests
+
+            tasks:
+            - id: test
+              type: io.kestra.core.services.PluginDefaultServiceTest$DefaultTester
+              set: 666""";
+
```

Comment:
If you want here is a test with a real task : 
https://github.com/kestra-io/kestra/pull/2454/files#diff-5bd1106f87ff619ff757907d0fe61eb03f8c2663727bf786814988e177edc229

Reply:
I added the test and reference the pull request to close it.
Thanks for point this.